apps/catalogue/src/views/HarmonizationRow.vue

```diff
@@ -0,0 +1,97 @@
+<template>
+  <tr>
+    <th class="table-label text-nowrap" scope="row">
+      {{ variable.name }}
+    </th>
+    <td
+      v-for="cohort in cohorts"
+      :key="cohort.acronym"
+      class="colored-grid-cell"
+      :class="'table-' + getCellClass(cohort)"
+    ></td>
+  </tr>
+</template>
+
+<script>
+import { mapActions } from "vuex";
+export default {
+  name: "HarmonizationRow",
+  props: {
+    variable: Object,
+    cohorts: Array,
+  },
+  data() {
+    return {
+      cohortMappings: undefined,
+    };
+  },
+  methods: {
+    ...mapActions(["fetchMappings"]),
+    getCellClass(cohort) {
+      return this.cohortMappings ? this.getMatchStatus(cohort) : null;
+    },
+    async fetchData() {
+      this.cohortMappings = await this.fetchMappings(this.variable);
+    },
+    getMatchStatus(cohort) {
+      if (this.variable.repeats) {
```

Comment:
Maybe extract this in a service or something or to the getters maybe? Later on for testing purposes it is easier when it is not in the vue component I guess? What do you think?

Reply:
I would extract it when I then indeed would need it seperate for testing. Interesting to observe how we seem to differ in our preferences. My preference is to keep things that belong together in one file as much as I can. You seem more at ease to working in multiple files. I think there is no objective truth so I propose this is up to the developer writing the code :-)

Reply:
i agree with both, 'keep things that belong together in one file' the same type of calc ( not the same) is used in other component. should probably be moved into service then these two functions are kept together. This PR is more explorative, trying to explore if the work around for missing argg data by paging the table work from a use perspective